### PR TITLE
BUG: retain tz in to_records

### PR DESCRIPTION
**Pull request by jbrockmendel**
plus the initial motivation: get rid of two of our `_internal_get_values` calls (of which i count 16 left in master)

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew

**Comment by jbrockmendel** [comment]
whatsnew added + green

File: pandas/tests/frame/methods/test_to_records.py [hunk 1]
```diff
@@ -18,6 +25,16 @@ def test_to_records_dt64(self):
         result = df.to_records()["index"][0]
         assert expected == result
 
+    def test_to_records_dt64tz_column(self):
+        df = DataFrame({"A": date_range("2012-01-01", "2012-01-02", tz="US/Eastern")})
```
Comment (jreback) [inline pandas/tests/frame/methods/test_to_records.py]: if you can add the issue numbe here

**Review by jreback — APPROVED** [review]
lgtm. merge on green after comment.

File: pandas/tests/frame/methods/test_to_records.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_to_records.py]: updated+green

**Pull request merged**